libsolidity/CMakeLists.txt

```diff
@@ -29,7 +29,15 @@ add_library(solidity ${sources} ${headers})
 target_link_libraries(solidity PUBLIC evmasm devcore ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY})
 
 if (${Z3_FOUND})
-  target_link_libraries(solidity PUBLIC ${Z3_LIBRARY})
+  if(SOLC_LINK_STATIC)
+	find_package(Threads)
+    set(Z3_LIBRARIES_EXTRA ${Z3_LIBRARIES_EXTRA} Threads::Threads)
+    find_package(OpenMP)
+	if(OpenMP_FOUND)
+		set(Z3_LIBRARIES_EXTRA ${Z3_LIBRARIES_EXTRA} OpenMP::OpenMP_CXX)
```

Comment:
Unfortunately, this only works starting from cmake 3.9 - with older cmake it'd be even messier...